activerecord/lib/active_record/connection_adapters/deduplicable.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module ConnectionAdapters # :nodoc:
+    module Deduplicable
+      extend ActiveSupport::Concern
+
+      module ClassMethods
+        def registry
+          @registry ||= {}
+        end
+
+        def new(*)
+          super.deduplicate
+        end
+      end
+
+      def deduplicate
+        self.class.registry[self] ||= self.deduplicated
```

Comment:
`self.deduplicated` can just be `deduplicated`, no? Also don't see why `deduplicated` should be a public method.

Reply:
Yeah good point, I missed that.